services/s3/src/test/java/software/amazon/awssdk/services/s3/internal/presignedurl/DefaultAsyncPresignedUrlManagerTest.java

```diff
@@ -0,0 +1,373 @@
+import software.amazon.awssdk.metrics.MetricPublisher;
+import software.amazon.awssdk.protocols.core.ExceptionMetadata;
+import software.amazon.awssdk.protocols.xml.AwsS3ProtocolFactory;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.InvalidObjectStateException;
+import software.amazon.awssdk.services.s3.model.NoSuchKeyException;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+import software.amazon.awssdk.services.s3.presignedurl.model.PresignedUrlGetObjectRequest;
+import software.amazon.awssdk.testutils.service.http.MockAsyncHttpClient;
+
+/**
+ * Tests for {@link DefaultAsyncPresignedUrlManager} using MockAsyncHttpClient to verify HTTP interactions.
+ */
+class DefaultAsyncPresignedUrlManagerTest {
+
+    private static final String TEST_CONTENT = "test-content";
+    private static final URI DEFAULT_ENDPOINT = URI.create("https://defaultendpoint.com");
+    private static final String TEST_URL = "https://test-bucket.s3.us-east-1.amazonaws.com/test-key?" +
+                                           "X-Amz-Date=20250707T000000Z&" +
+                                           "X-Amz-Signature=test-signature-value&" +
+                                           "X-Amz-Algorithm=AWS4-HMAC-SHA256&" +
+                                           "X-Amz-SignedHeaders=host&" +
+                                           "X-Amz-Security-Token=test-session-token&" +
+                                           "X-Amz-Credential=AKIAIOSFODNN7EXAMPLE%2F20250707%2Fus-east-1%2Fs3%2Faws4_request&" +
+                                           "X-Amz-Expires=86400";
+
+    private MockAsyncHttpClient mockHttpClient;
+    private DefaultAsyncPresignedUrlManager presignedUrlManager;
+    private PresignedUrlGetObjectRequest testRequest;
+    private AwsProtocolMetadata protocolMetadata;
+    private AwsS3ProtocolFactory protocolFactory;
+    private AsyncClientHandler clientHandler;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        mockHttpClient = new MockAsyncHttpClient();
+        URL testPresignedUrl = new URL(TEST_URL);
+        testRequest = PresignedUrlGetObjectRequest.builder()
+                                                  .presignedUrl(testPresignedUrl)
+                                                  .build();
+
+        SdkClientConfiguration clientConfiguration = getDefaultSdkConfigs();
+        protocolMetadata = AwsProtocolMetadata.builder()
+                                              .serviceProtocol(AwsServiceProtocol.REST_XML)
+                                              .build();
+        protocolFactory = initProtocolFactory(clientConfiguration);
+        clientHandler = new AwsAsyncClientHandler(clientConfiguration);
+
+        presignedUrlManager = new DefaultAsyncPresignedUrlManager(
+            clientHandler, protocolFactory, clientConfiguration, protocolMetadata);
```

Comment:
Once it's implemented, we should test creation of this from the API that the customer uses. When we do that, we need to test all of the configuration overrides they may be trying. Retries, user agent, etc. That's probably dozens of tests, but it'll be important to make sure all of the functionality works and is safe against regression.